src/hotspot/os_cpu/bsd_aarch64/safefetch_bsd_aarch64.S

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2022 SAP SE. All rights reserved.
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifdef __APPLE__
+# Darwin uses _ prefixed global symbols
+#define SYMBOL(s) _ ## s
+#define ELF_TYPE(name, description)
+#else
+#define SYMBOL(s) s
+#define ELF_TYPE(name, description) .type name,description
+#endif
+
+    .global SYMBOL(SafeFetchN)
+    .global SYMBOL(_SafeFetchN_fault)
+    .global SYMBOL(_SafeFetchN_continuation)
+    .global SYMBOL(SafeFetch32)
+    .global SYMBOL(_SafeFetch32_fault)
+    .global SYMBOL(_SafeFetch32_continuation)
+
+    # Support for int SafeFetch32(int* address, int defaultval);
+    #
+    #  x0 : address
+    #  w1 : defaultval
+
+	# needed to align function start to 4 byte
+    .align  6
+    ELF_TYPE(SafeFetch32,@function)
+SYMBOL(SafeFetch32):
+SYMBOL(_SafeFetch32_fault):
+    ldr      w1, [x0]
+SYMBOL(_SafeFetch32_continuation):
+    mov      x0, x1
+    ret
```

Comment:
Minor: on RISC platforms you seems to prefer to pack exceptional and non-exceptional paths, and on x86 they are separated. If we align this with x86, we'd save an instruction in each execution of non-exceptional path, with an extra instruction in static code.
```
    ldr      w0, [x0]
    ret
SYMBOL(_SafeFetch32_continuation):
    mov      x0, x1
    ret
```

Reply:
I'm surprised this would work. Can you really specify the same register as load destination and as load source?

Reply:
Ah, it seems to work. You live and learn :)

Reply:
Yes, on arm32/64 it would not work if the instruction uses a register as a target and write-back, like `ldr r0, [r0, 4]!`, otherwise it's OK. AFAIR, it's also OK on ppc. 

As a reference, I've checked the output of gcc/clang, both don't use the extra register, so the argument regarding performance looks valid :) Not insisting on fixing, this is indeed minor.

```
anton@mercury:/tmp$ cat test.c
long test(long* addr) {
	return *addr;
}
anton@mercury:/tmp$ aarch64-linux-gnu-gcc --version
aarch64-linux-gnu-gcc (GCC) 11.2.0
anton@mercury:/tmp$ aarch64-linux-gnu-gcc -c -O3 -o test.o test.c
anton@mercury:/tmp$ aarch64-linux-gnu-objdump -d test.o

0000000000000000 <test>:
   0:	f9400000 	ldr	x0, [x0]
   4:	d65f03c0 	ret
```